File: Python/747. Largest Number At Least Twice of Others.py
class Solution:
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ma, mai = -1, -1
        mma = -1
        for i in range(len(nums)):
            if nums[i] > ma:
                ma = nums[i]
                mai = i
        for i in range(len(nums)):
            if nums[i] > mma and nums[i] != ma:
                mma = nums[i]
        if ma >= 2*mma:
            return mai
        return -1

File: C++/747. Largest Number At Least Twice of Others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int ma = -1,ans = -1,sec = -1;
        int len = nums.size();
        for(int i=0;i<len;i++)
        {
            if(nums[i]>ma)
            {
                sec = ma;
                ma = nums[i];
                ans = i;
            }
            else if(nums[i]>sec)
                sec = nums[i];
        }
        printf("%d %d\n",ma, sec);
        return ma >= 2*sec?ans:-1;
    }
};